Kernel/Memory/MemoryManager.cpp

```diff
@@ -462,25 +464,25 @@ UNMAP_AFTER_INIT void MemoryManager::parse_memory_map_fdt(MemoryManager::GlobalD
                     state.state = State::Root;
                     break;
                 case State::InMemory:
-                    VERIFY(state.start.has_value() && state.size.has_value());
-                    dbgln("MM: Memory Range {}: address: {} size {:#x}", node_name, PhysicalAddress { state.start.value() }, state.size.value());
-                    global_data.physical_memory_ranges.append(PhysicalMemoryRange { PhysicalMemoryRangeType::Usable, PhysicalAddress { state.start.value() }, state.size.value() });
+                    for (auto const& reg_entry : state.reg) {
```

Comment:
The preallocating might be possible here as well, as well as in the ReservedMemory one